### PR TITLE
vapor: Fix the build and update: Use correct deps and find numpy incdir

### PR DESCRIPTION
**Pull request by bernhardkaindl**
vapor needs `proj@:7` and gives a list of tested dependency versions.
Make it find the `numpy` include path and add version 3.5.0 as well

### CODE REVIEW

File: var/spack/repos/builtin/packages/vapor/package.py
Comment (RemiLacroix-IDRIS) [inline var/spack/repos/builtin/packages/vapor/package.py:22]: @bernhardkaindl: Just double-checking, did the checksum for version 3.3.0 really change?

File: var/spack/repos/builtin/packages/vapor/package.py
Comment (RemiLacroix-IDRIS) [inline var/spack/repos/builtin/packages/vapor/package.py:52]: If that's specific to Numpy, maybe it would make sense to use `NUMPY_INCLUDE_DIR`?

**Comment by RemiLacroix-IDRIS** [comment]
@bernhardkaindl: Have you seen the two comments I made?

**Comment by bernhardkaindl** [comment]
> @bernhardkaindl: Have you seen the two comments I made?

@RemiLacroix-IDRIS No, they are not here, and the only email notification about vapor that I have is this comment. Can you add them again?

**Comment by RemiLacroix-IDRIS** [comment]
@bernhardkaindl: Don't you see them when going on https://github.com/spack/spack/pull/26630/files?

See this screenshot: 
image

**Comment by bernhardkaindl** [comment]
@RemiLacroix-IDRIS Yes it did change:
```py
spack checksum vapor@3.3.0
==> Found 1 version of vapor:
  
  3.3.0  https://github.com/NCAR/VAPOR/archive/3.3.0.tar.gz

==> Fetching https://github.com/NCAR/VAPOR/archive/3.3.0.tar.gz

    version('3.3.0', sha256='41c13d206cfcfa4146155d524106de2eb74e7b59af1e2f8c1c3056c15d508a93')
```

File: var/spack/repos/builtin/packages/vapor/package.py
Comment (bernhardkaindl) [inline var/spack/repos/builtin/packages/vapor/package.py:22]: Yes it did change.

**Comment by bernhardkaindl** [comment]
> @bernhardkaindl: Don't you see them when going on https://github.com/spack/spack/pull/26630/files?

@RemiLacroix-IDRIS No. It would be normal in case you made the comments but didn't submit your review. 

In this case, the green button labelled "Review Changed" on the top-right corner would have a bright-green circle which contains the number "2" in it, for the two comments that weren't submitted yet.
If it looks like this submit them, you'd have to click the button and press the green button "Submit review" in the popup dialogue which appears below then. It's at the lower border of that window, so its easy to miss.

**Review by RemiLacroix-IDRIS — CHANGES_REQUESTED** [review]

**Comment by RemiLacroix-IDRIS** [comment]
@bernhardkaindl: Sorry, I didn't understand I had to submit the review after adding the comments!

File: var/spack/repos/builtin/packages/vapor/package.py
Comment (bernhardkaindl) [reply var/spack/repos/builtin/packages/vapor/package.py:52]: I tried (now again) but `NUMPY_INCLUDE_DIR` didn't work.
I looked into it but could not understand why.

Comment (RemiLacroix-IDRIS) [reply var/spack/repos/builtin/packages/vapor/package.py:52]: Ok.

**Review by RemiLacroix-IDRIS — APPROVED** [review]
LGTM

**Comment by bernhardkaindl** [comment]
> LGTM

Thanks, I had another go at the numpy include. I found the issue now:
```cmake
include_directories (${NUMPY_INCLUDE_DIR})
```
.. is only done inside `if (WIN32)` :-1:  The workaround works.

@RemiLacroix-IDRIS :+1: 

Can you try to make a formal review by selecting **Commits** or **Changed Files** and use the **Review Changes** button to submit your review, selecting the **Approve** radio button before clicking **Submit review** below it?

If you have permission for it (it's not grey anywhere), then we can merge with your submitted approval afterwards.

**Comment by RemiLacroix-IDRIS** [comment]
> Can you try to make a formal review by selecting Commits or Changed Files and use the Review Changes button to submit your review, selecting the Approve radio button before clicking Submit review below it?

I think it's what I did earlier.

As far as I know, I'm not authorized to merge PRs.

**Review by RemiLacroix-IDRIS — APPROVED** [review]

**Comment by bernhardkaindl** [comment]
@sethrj I'd need an approval from a reviewer witth write access, RemiLacroix-IDRIS who is the maintainer of vapor gave his approval (twice, actually - sorry!)

**Comment by sethrj** [comment]
OK, I'll add my approval, but @RemiLacroix-IDRIS you should do everything in your power to avoid retagging different state points of your code as the same version. Changing SHA also could be a sign of nefarious activity...

**Comment by sethrj** [comment]
It also breaks the build for anyone who already has installed this version in the past, because the checksum no longer matches.

**Review by sethrj — APPROVED** [review]

**Comment by RemiLacroix-IDRIS** [comment]
@sethrj: That's not my code so I have no power at all. :) I did check the official release page and they indeed seem to have changed the archive file for 3.3.0, no idea why though.

**Comment by sethrj** [comment]
OK thanks @RemiLacroix-IDRIS , I misinterpreted "RemiLacroix-IDRIS who is the maintainer of vapor" as being the code, not the spack package :)  Thanks for verifying.

**Comment by samfux84** [comment]
This PR also made a change to the glew package which is not discussed at all. Why was the "gl" dependency changed to "glu" in the glew package?

**Comment by RemiLacroix-IDRIS** [comment]
> This PR also made a change to the glew package which is not discussed at all. Why was the "gl" dependency changed to "glu" in the glew package?

As far as I can tell "glew" does depend on "glu" which itself depends on "gl". The dependency was wrong before.

**Comment by bernhardkaindl** [comment]
> > This PR also made a change to the glew package which is not discussed at all. Why was the "gl" dependency changed to "glu" in the glew package?
> 
> As far as I can tell "glew" does depend on "glu" which itself depends on "gl". The dependency was wrong before.

This was also my impression, http://glew.sourceforge.net/log.html has: "Define GLEW_NO_GLU for no glu dependency.",
so this may be ancient history.

As far as memory serves, my build missed `glu.h`, which was cured by this change. I am trying to verify this now:

So far, I was able to build `glew` using `depends_on("gl")` (instead of `glu`), now building `vapor` with that.

**Pull request merged**
